benchmarks/performance-poetry/complex-poetry/src/main/java/com/squareup/benchmarks/performance/complex/poetry/PerformancePoemWorkflow.kt

```diff
@@ -89,39 +93,73 @@ class PerformancePoemWorkflow(
   ): OverviewDetailScreen {
     return when (renderState) {
       Initializing -> {
-        // Again, then entire `Initializing` state is a smell, which is most obvious from the
+        // Again, the entire `Initializing` state is a smell, which is most obvious from the
         // use of `Worker.from { Unit }`. A Worker doing no work and only shuttling the state
         // along is usually the sign you have an extraneous state that can be collapsed!
         // Don't try this at home.
-        context.runningWorker(Worker.from { }, "initializing") {
-        isLoading.value = true
-        action {
-          isLoading.value = false
-          state = Selected(NO_SELECTED_STANZA)
+        context.runningWorker(
+          Worker.from {
+            isLoading.value = true
+          },
+          "initializing"
+        ) {
+          action {
+            isLoading.value = false
+            state = Selected(NO_SELECTED_STANZA)
+          }
         }
-      }
         OverviewDetailScreen(overviewRendering = BackStackScreen(BlankScreen))
       }
       else -> {
-        val (stanzaIndex, currentStateIsLoading) = when (renderState) {
-          is ComplexCall -> Pair(renderState.payload, true)
-          is Selected -> Pair(renderState.stanzaIndex, false)
+        val (stanzaIndex, currentStateIsLoading, repeat) = when (renderState) {
+          is ComplexCall -> Triple(renderState.payload, true, renderState.repeater)
+          is Selected -> Triple(renderState.stanzaIndex, false, 0)
           Initializing -> throw IllegalStateException("No longer initializing.")
         }
 
         if (currentStateIsLoading) {
-          context.runningWorker(
-            TraceableWorker.from("PoemLoading") {
-              isLoading.value = true
-              delay(simulatedPerfConfig.complexityDelay)
-              // No Output for Worker is necessary because the selected index
-              // is already in the state.
+          if (repeat > 0) {
+            // Running a flow that emits 'repeat' number of times
+            context.runningWorker(
+              flow {
+                while (repeat > 0) {
```

Comment:
Because of the check above, this is always true.

Reply:
good point, but I think this is clearer than `while (true)`. Maybe not.

The point is the flow needs to keep running and emitting as long as we are in this state

Reply:
Updated. 

```
 if (repeat > 0) {
            // Running a flow that emits 'repeat' number of times
            context.runningWorker(
              flow {
                while (true) {
                  // As long as this Worker is running we want to be emitting values.
                  delay(2)
                  emit(repeat)
                }
              }.asTraceableWorker("EventRepetition")
            ) {
```

WDYT @RBusarow ?

Reply:
LGTM!